docs/source/cc_launcher.md

```diff
@@ -0,0 +1,301 @@
+- `client_cert`: The PEM encoded TLS client certificate generated by the peer
+  that must be used when the chaincode establishes its connection to the peer.
+- `client_key`: The PEM encoded client key generated by the peer that must be
+  used when the chaincode establishes its connection to the peer.
+- `root_cert`: The PEM encoded TLS root certificate for the `ChaincodeSupport`
+  gRPC server endpoint hosted by the peer.
+
+When `run` terminates, the peer considers the chaincode terminated. If another
+request arrives for the chaincode, the peer will attempt to start another
+instance of the chaincode by invoking `run` again. The contents of
+`chaincode.json` must not be cached across invocations.
+
+The following is an example of a simple `run` script for go chaincode:
+```sh
+#!/bin/bash
+
+BUILD_OUTPUT_DIR="$1"
+RUN_METADATA_DIR="$2"
+
+# setup the environment expected by the go chaincode shim
+export CORE_CHAINCODE_ID_NAME="$(jq -r .chaincode_id "$RUN_METADATA_DIR/chaincode.json")"
+export CORE_PEER_TLS_ENABLED="true"
+export CORE_TLS_CLIENT_CERT_FILE="$RUN_METADATA_DIR/client.crt"
+export CORE_TLS_CLIENT_KEY_FILE="$RUN_METADATA_DIR/client.key"
+export CORE_PEER_TLS_ROOTCERT_FILE="$RUN_METADATA_DIR/root.crt"
+
+# populate the key and certificate material used by the go chaincode shim
+jq -r .client_cert "$RUN_METADATA_DIR/chaincode.json" > "$CORE_TLS_CLIENT_CERT_FILE"
+jq -r .client_key  "$RUN_METADATA_DIR/chaincode.json" > "$CORE_TLS_CLIENT_KEY_FILE"
+jq -r .root_cert   "$RUN_METADATA_DIR/chaincode.json" > "$CORE_PEER_TLS_ROOTCERT_FILE"
+if [ -z "$(jq -r .client_cert "$RUN_METADATA_DIR/chaincode.json")" ]; then
+    export CORE_PEER_TLS_ENABLED="false"
+fi
+
+# exec the chaincode to replace the script with the chaincode process
+exec "$BUILD_OUTPUT_DIR/chaincode" -peer.address="$(jq -r .peer_address "$ARTIFACTS/chaincode.json")"
+```
+
+## Configuring External Builders and Launchers
+
+Configuring the peer to use external builders involves adding a configuration
+block to `core.yaml` that defines external builders. Each external builder
+definition must include a name (used for logging) and the path to parent of
+the `bin` directory containing the builder scripts.
+
+An optional list of environment variable names to propagate from the peer when
+invoking the external builder scripts can also be provided.
+
+The following example defines two external builders:
+
+```yaml
+chaincode:
+  externalBuilders:
+  - name: my-golang-builder
+    path: /builders/golang
+    environmentWhitelist:
+    - GOPROXY
+    - GONOPROXY
+    - GOSUMDB
+    - GONOSUMDB
+  - name: noop-builder
+    path: /builders/binary
+```
+
+In this example, the implementation of "my-golang-builder" is contained within
+the `/builders/golang` directory and its build scripts are located in
+`/builders/golang/bin`. When the peer invokes any of the build scripts
+associated with "my-golang-builder", it will not propagate the values of any
```

Comment:
Per Jason's similar comment above I changed it to:
```
When the peer invokes any of the build scripts
associated with "my-golang-builder", it will propagate only the values of the
environment variables in the whitelist.

Note: The following environment variables are always propagated to external
builders:

- LD_LIBRARY_PATH
- LIBPATH
- PATH
- TMPDIR
```

Let me know if that is not sufficient.